Assets/HoloToolkit/SpatialMappingComponent/Support/SMBaseAbstract.cs

```diff
@@ -0,0 +1,600 @@
+    /// <param name="changeType">What type of change this is (add, update, or remove)</param>
+    /// <param name="bounds">The bounds of the mesh</param>
+    /// <param name="updateTime">The time the update occurred</param>
+    protected virtual void SurfaceObserver_OnSurfaceChanged(SurfaceId surfaceId, SurfaceChange changeType, Bounds bounds, System.DateTime updateTime)
+    {
+        switch (changeType)
+        {
+            case SurfaceChange.Added:
+            case SurfaceChange.Updated:
+                HandleAdd(surfaceId, updateTime, bakeMeshes);
+                break;
+            case SurfaceChange.Removed:
+                HandleDelete(surfaceId);
+                break;
+            default:
+                break;
+        }
+    }
+
+    /// <summary>
+    /// Handles cleaning up a known mesh or moving it to the RemovedMeshObjects list
+    /// 
+    /// If NumUpdatesBeforeRemoval < 1, the mesh will immediately be destroyed upon removal
+    /// Else we will create a new RemoveSurfaceHolder and add that to the RemovedMeshObjects list to cache the mesh until we believe it should actually be removed
+    /// </summary>
+    /// <param name="surfaceId"></param>
+    protected virtual void HandleDelete(SurfaceId surfaceId)
+    {
+        GameObject obj = SpatialMeshObjects[surfaceId];
+        SpatialMeshObjects.Remove(surfaceId);
+
+        if (NumUpdatesBeforeRemoval < 1)
+        {
+            FinalDestroy(obj);
+        }
+        else if (obj != null)
+        {
+            bool shouldDisable = !ShouldBeActiveWhileRemoved(obj);
+
+            MeshCollider collider = obj.GetComponent<MeshCollider>();
+            bool collisionsEnabled = (collider != null && collider.enabled);
+            if (collisionsEnabled && shouldDisable)
+            {
+                collider.enabled = false;
+            }
+
+            MeshRenderer mr = obj.GetComponent<MeshRenderer>();
+            bool rendererEnabled = (mr != null && mr.enabled);
+            if (rendererEnabled && shouldDisable)
+            {
+                mr.enabled = false;
+            }
+            RemovedSurfaceHolder holder = new RemovedSurfaceHolder(NumUpdatesBeforeRemoval + 1 /*+1 because we are going to process it now*/, obj, surfaceId, collisionsEnabled, rendererEnabled);
+            RemovedMeshObjects.Add(surfaceId, holder);
+        }
+    }
+
+    /// <summary>
+    /// Destroys the provided GameObject and The MeshFilter.mesh if present
+    /// </summary>
+    /// <param name="surfaceObject">The GameObject to destroy</param>
+    protected void FinalDestroy(GameObject surfaceObject)
+    {
+        if (surfaceObject != null)
+        {
+            MeshFilter filter = surfaceObject.GetComponent<MeshFilter>();
+            if (filter != null)
+            {
+                if (filter.mesh != null)
+                {
+                    Destroy(filter.mesh);
+                }
+            }
+
+            Destroy(surfaceObject);
```

Comment:
I feel Destroy is more appropriate here. We don't have a need to remove these objects this instant (as in before the end of the frame) and Destroy is generally seen as safer